refactor: remove `randSeedWeightedItem`

DayKev opened the pull request:
## What are the changes the user will see?
Slightly more random trainer parties?

## Why am I making these changes?
We added a replacement for `randSeedWeightedItem` but didn't remove it. Now it's gone.

## What are the changes from a developer perspective?
The single use of `randSeedWeightedItem` was replaced with `randSeedItem`. A very minor cleanup of the `Trainer` class constructor was also done.

## Checklist
- [x] Otherwise: **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?

DayKev committed:
refactor: remove `randSeedWeightedItem`

Also minor cleanup of `Trainer` class constructor

MokaStitcher reviewed (commented):
> We added a replacement for randSeedWeightedItem but didn't remove it. Now it's gone.

What's the replacement?

RNG.weightedPick favors the elements at the beginning of the array [as per Phaser documentation](https://docs.phaser.io/api-documentation/class/math-randomdatagenerator), so even if it wasn't documented by us we know what it did and it's not the same as randSeedItem.

If everyone is fine with the change though 🤷

DayKev commented:
> > We added a replacement for randSeedWeightedItem but didn't remove it. Now it's gone.
> 
> What's the replacement?

https://github.com/Despair-Games/poketernity/blob/82ebca528e009d12a9ebed9a8b46dfa8f08a5b6c/src/utils/random-utils.ts#L93-L110

Added in #898

> RNG.weightedPick favors the elements at the beginning of the array [as per Phaser documentation](https://docs.phaser.io/api-documentation/class/math-randomdatagenerator), so even if it wasn't documented by us we know what it did and it's not the same as randSeedItem.

Well it just says "favoring the earlier entries" but that doesn't tell us *how* it does that. The new function lets us explicitly set the weights we want.

DayKev committed:
Merge branch 'beta' into refactor/remove-randseedweighteditem

MokaStitcher reviewed (approved):
> Well it just says "favoring the earlier entries" but that doesn't tell us how it does that. The new function lets us explicitly set the weights we want.

Ok but this PR doesn't use that new function in the spot where it removes the old one. Whether we want to keep the same or similar behavior is a game design decision. I don't know how much it impacts matters so :shrug:

DayKev commented:
> > Well it just says "favoring the earlier entries" but that doesn't tell us how it does that. The new function lets us explicitly set the weights we want.
> 
> Ok but this PR doesn't use that new function in the spot where it removes the old one. Whether we want to keep the same or similar behavior is a game design decision. I don't know how much it impacts matters so 🤷

We might want to use weighted party templates, but the current ones weren't designed with the understanding that the weighted randomness was even being used here so maybe not.
